lib/cisco_node_utils/interface.rb

```diff
@@ -499,13 +475,11 @@ def default_mtu
     end
 
     def speed
-      config_get('interface', 'speed', name: @name)
+      val = config_get('interface', 'speed', name: @name)
+      val == 'auto' ? val : val.to_i
```

Comment:
It doesn't *really* matter but my thinking is:
  * The config_get's are usually dealing with strings (as a result of the regex matches)
  * On the puppet side the manifest can specify either but the type should normalize / munge it regardless
  * If we consistently treat these as strings then we don't have to coerce them (other than in the type munge)